library/Icingadb/Widget/Detail/ObjectDetail.php

```diff
@@ -494,4 +498,36 @@ protected function getUsersAndUsergroups()
 
         return [$users, $usergroups];
     }
+
+    /**
+     * Get all action urls configured for $this->object
+     *
+     * @return array
+     */
+    protected function getActionUrls(): array
+    {
+        if ($this->object->action_url) {
+            //NOTE: action_url of $this->object has a relation to the `ActionUrl` class
+            //      that's why we access it in such a nested way
+            return $this->object->action_url->action_url;
+        }
+
+        return [];
+    }
+
+    /**
+     * Get all note urls configured for $this->object
+     *
+     * @return array
+     */
+    protected function getNotesUrls(): array
+    {
+        if ($this->object->notes_url) {
+            //NOTE: notes_url of $this->object has a relation to the `NotesUrl` class
+            //      that's why we access it in such a nested way
+            return $this->object->notes_url->notes_url;
+        }
+
+        return [];
+    }
```

Comment:
We require PHP 7 now for Icinga DB Web. So it's possible to use the new null coalescing operator (`??`). This allows the following:

```
foreach ($this->object->action_url->action_url ?? [] as $url) {
    ...
}
```